Swift_basics_hometask_02.playground/Contents.swift

```diff
@@ -3,84 +3,229 @@ import UIKit
+
+dates.forEach { (key, value) in
+    print(key, value)
+}
 
+dates.keys.forEach { print($0) }
+dates.values.forEach { print($0) }
+
+for (month, day) in dates {
+    guard day == 30 else { continue }
+    print(month)
+}
+
+for month in dates.keys {
+    guard month.count < 7 else { continue }
+    print(month)
+}
+
+// MARK: - Task 3
+print("Task 3 ...")
 // Создайте словарь как журнал студентов, где имя и фамилия студента - это ключ, а оценка - это значение.
 // Некоторым студентам измените текущие оценки.
 // Затем добавьте 3х новых студентов в группу и оцените их.
 // А потом удалите 5х ребят, потому что они отчислились.
 // В конце посчитайте общий бал и средний бал ребят, которые остались в группе.
 
-// MARK: - Task 4
+var grates: [String: Int] = ["Ilya Pastrevich": 5, "Anton Pastrevich": 1, "Nikita Litvinenok": 2, "Kate Dashkevich": 3]
+grates["Anton Pastrevich"] = 2
+grates["Nikita Litvinenok"] = 1
+
+grates["Sasha Sinkevich"] = 3
+grates["Valik Henin"] = 5
+grates["Kolya Kozlovsky"] = 2
 
+grates["Anton Pastrevich"] = nil
+grates["Nikita Litvinenok"] = nil
+grates["Sasha Sinkevich"] = nil
+grates["Valik Henin"] = nil
+grates["Kolya Kozlovsky"] = nil
+
+var sum = 0
+grates.values.forEach { sum += $0 }
+print(sum)
+print(sum / grates.count)
+
+// MARK: - Task 4
+print("Task 4 ...")
 // Создайте словарь типа [String: (Int, Int)?],
 // Где ключ - это String и хранит в себе название шахмотной фигуры
 // А значение - это tuple и хранит в себе координаты фигуры X и Y на шахматной доске.
 // Обратите внимание, что значение tuple - опционал и говорит о том, что фигуру "убили".
 // Добавьте 5 разных фигур на шахматную доску и выведите в консоль информацию о положении каждой фигуры или ее отсутствии на поле.
 
+var game: [String: (Int, Int)?] = ["King": (1, 6), "Qeen": nil, "Bishop": (4, 7), "Knight": nil, "Rook": (2, 9), "Pawns": nil]
+game["KingX"] = nil
+game["QeenX"] = (6, 1)
+game["BishopX"] = nil
+game["RookX"] = (8, 9)
+game["PawnsX"] = nil
+for (figure, position) in game {
+    print("Name: \(figure) Position: \(String(describing: position))")
+}
+
 // MARK: - Task 5.
+print("Task 5 ...")
 // Создайте кортеж с тремя параметрами: ваш любимый фильм, ваше любимое число и ваше любимое блюдо.
 //  Все элементы кортежа должны быть именованы.
 // Одним выражением запишите каждый элемент кортежа в три константы.
 // Теперь создайте второй кортеж, аналогичный первому по параметрам, но описывающий другого человека.
 // Создайте новый кортеж, элементами которого будут любимое число из первого кортежа, любимое число из второго кортежа и разница любимых чисел первого и второго кортежей.
 // Решение оформите ниже
 
+let favorite = ("Tor", 13, "Meat")
+let (favoriteFilm, favoriteNumber, favoriteFood) = favorite
+print("Film \(favoriteFilm)")
+let favoriteFriend = ("Madmen", 10, "Potato")
+let (favoriteFilmFriend, favoriteNumberFriend, favoriteFoodFriend) = favoriteFriend
+let favoriteNumbers = (13, 10, favoriteNumber - favoriteNumberFriend)
```

Comment:
what is first fav number will be 123 and for second 0
u need to use values from 1st and 2nd tuples